src/blosc2/proxy.py

```diff
@@ -296,6 +292,58 @@ async def afetch(self, item: slice | list[slice] = None) -> blosc2.NDArray | blo
         -----
         This method is only available if the :ref:`ProxySource` or :ref:`ProxyNDSource`
         have an async `aget_chunk` method.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import blosc2
+        >>> import asyncio
+        >>> class MyProxySource:
+        >>>     def __init__(self, data, chunk, block):
+        >>>             # If the next source is multidimensional, it must have the attributes:
+        >>>             self.data = data
+        >>>             self.shape = data.shape
+        >>>             self.chunks = chunk
+        >>>             self.blocks = block
+        >>>             self.dtype = data.dtype
+        >>>             f"Data shape: {self.shape}, Chunks: {self.chunks}"
+        Data shape: (4, 5), Chunks: [2, 2]
+        >>>             f"Blocks: {self.blocks}, Dtype: {self.dtype}"
+        Blocks: [2, 2], Dtype: int64
+        >>> # This method must be present.
+        >>>     async def aget_chunk(self, nchunk):
+        >>>             await asyncio.sleep(0.1)
+        >>>             return self.data[nchunk]
+        >>> class MyProxy(blosc2.Proxy):
+        >>>     def __init__(self, source):
+        >>>               super().__init__(source)
+        >>>               self._cache = source.data
+        >>>     # Asynchronous method to get the cache data
+        >>>     async def afetch(self, slice_=None):
+        >>>               return self._cache if slice_ is None else self._cache[slice_]
```

Comment:
You do not need to create a class `MyProxy` which inherits from `blosc2.Proxy`. You can directly create the proxy with `proxy = blosc2.Proxy(source)`